client/app/bundles/comments/startup/ServerApp.jsx

```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import { Provider } from 'react-redux';
-
-import createStore from '../store/commentsStore';
 import NonRouterCommentsContainer from '../containers/NonRouterCommentsContainer';
+import NavigationBarContainer from '../containers/NavigationBarContainer';
 
-export default props => {
-  const store = createStore(props);
+export default (_props, _railsContext) => {
+  const store = ReactOnRails.getStore('commentsStore');
   return (
     <Provider store={store}>
+      <NavigationBarContainer />
```

Comment:
Turns out that it was Provider! @radixhound and @robwise. 

Provider.js has this React call:

```
React.Children.only
object React.Children.only(object children)
Return the only child in children. Throws otherwise. 
```

I had a bug where there server app (but *NOT* the client app) had an extra component as pointed out above, and if the the server is not throwing errors, then the error does not show in the tests!

@robwise @mapreal19 @radixhound @jbhatab Please provide final review!